src/pages/sales/SalesOrderDetails.tsx

```diff
@@ -68,56 +76,129 @@ const steps = [
 
 const SalesOrderDetails = () => {
   let params = new URLSearchParams(window.location.search);
+  const id = params.get('id');
   const navigate = useNavigate();
-  //For now, will fetch from context, in future, will use API call
-  const { salesOrders } = React.useContext(salesContext);
+  const { products } = React.useContext(inventoryContext);
   const [salesOrder, setSalesOrder] = useState<SalesOrder>();
-  const [loading, setLoading] = React.useState<boolean>(true);
+  const [editSalesOrderItems, setEditSalesOrderItems] = useState<
+    SalesOrderItem[]
+  >([]);
+  const [newSalesOrderItem, setNewSalesOrderItem] = useState<SalesOrderItem>();
+  const [loading, setLoading] = useState<boolean>(true);
   const [alert, setAlert] = useState<AlertType | null>(null);
-  const [activeStep, setActiveStep] = React.useState<number>(0);
+  const [activeStep, setActiveStep] = useState<number>(0);
+  const [showDialog, setShowDialog] = useState<boolean>(false);
+  const [availProducts, setAvailProducts] = useState<Product[]>([]);
 
-  const id = params.get('id');
+  const columns: GridColDef[] = [
+    {
+      field: 'productName',
+      headerName: 'Product Name',
+      flex: 2,
+      valueGetter: (params) => params.row.productName
+    },
+    {
+      field: 'quantity',
+      type: 'number',
+      headerName: 'Quantity',
+      flex: 1
+    },
+    {
+      field: 'price',
+      type: 'number',
+      headerName: 'Price',
+      flex: 1,
+      valueGetter: (params) => params.row.price ?? 0,
+      valueFormatter: (params) => params.value.toFixed(2)
+    },
+    {
+      field: 'action',
+      headerName: 'Action',
+      headerAlign: 'center',
+      align: 'center',
+      flex: 1,
+      renderCell: (params) => {
+        if (
+          params.row.isNewAdded &&
+          salesOrder?.orderStatus === OrderStatus.PREPARING
+        ) {
+          return (
+            <>
+              <Button
+                variant='contained'
+                size='medium'
+                onClick={() => {
+                  removeItemFromList(params.row.productName);
+                }}
+              >
+                Remove Item
+              </Button>
+            </>
+          );
+        }
+      }
+    }
+  ];
 
   useEffect(() => {
     setLoading(true);
-    const saleOrd = salesOrders.find((order) => {
-      return order.id === parseInt(id!);
-    });
+    id &&
+      asyncFetchCallback(
+        getSalesOrderDetailsSvc(id),
+        (salesOrder: SalesOrder) => {
+          if (salesOrder) {
+            setSalesOrder(salesOrder);
+            setEditSalesOrderItems([...salesOrder.salesOrderItems]);
+            setAvailProducts(
+              products.filter((product) => {
+                return salesOrder?.salesOrderItems.find(
+                  (orderItem) => orderItem.productName !== product.name
+                );
+              })
+            );
+            switch (salesOrder.orderStatus) {
```

Comment:
this whole switch statement can be refactored by including the OrderStatus as a value in each object within the steps array, then finding the index of the object that matches the value

`const steps = [{
    label: 'Order Placed',
    value: OrderStatus.CREATED,
    icon: <ReceiptLongRounded sx={{ fontSize: 35 }} />,
    nextAction: 'Confirm Payment'
  },...`

then afterwards,
```suggestion
            setActiveStep(steps.findIndex(step => step.value === salesOrder.orderStatus)
```